fineract-provider/src/main/java/org/apache/fineract/portfolio/delinquency/service/DelinquencyWritePlatformServiceImpl.java

```diff
@@ -134,8 +144,61 @@ public CommandProcessingResult deleteDelinquencyBucket(Long delinquencyBucketId,
                         "Data integrity issue with resource: " + delinquencyBucket.getId());
             }
             repositoryBucket.delete(delinquencyBucket);
+            return new CommandProcessingResultBuilder().withCommandId(command.commandId()).withEntityId(delinquencyBucket.getId()).build();
         }
-        return new CommandProcessingResultBuilder().withCommandId(command.commandId()).withEntityId(delinquencyBucket.getId()).build();
+        return new CommandProcessingResultBuilder().withCommandId(command.commandId()).withEntityId(delinquencyBucketId).build();
+    }
+
+    @Override
+    public LoanScheduleDelinquencyData getOverdueDays(LoanScheduleDelinquencyData loanScheduleDelinquencyData) {
```

Comment:
This is definitely not gonna fly. We're introducing side effects on an input parameter. Please refactor this in a way that it's returning an object from the inside.

Reply:
The idea to pass as parameter and return the DTO is because in that function we add the Loan entity that is used in different part and avoid the Loan read n times as It was previously